File: JavaScript/876. Middle of the Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    var left = head;
    var right = head;
    while(right!= null && right.next!=null){
        left= left.next;
        right= right.next.next;
    }
    return left;
};

File: C/876. Middle of the Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head){
    struct ListNode* left = head;
    struct ListNode* right = head;
    while(right!= NULL && right->next!=NULL){
        left= left->next;
        right= right->next->next;
    }
    return left;
}
